drivers/dac/ad3552r/ad3552r.c

```diff
@@ -1209,16 +1253,30 @@ int32_t ad3552r_reset(struct ad3552r_desc *desc)
 	return _ad3552r_set_reg_attr(desc, AD3552R_ADDR_ASCENSION, 0);
 }
 
-int32_t ad3552r_ldac_trigger(struct ad3552r_desc *desc, uint16_t mask)
+int32_t ad3552r_ldac_trigger(struct ad3552r_desc *desc, uint16_t mask,
+			     uint8_t is_fast)
 {
-	if (!desc->ldac)
-		return ad3552r_write_reg(desc, AD3552R_REG_ADDR_SW_LDAC_24B,
-					 mask);
+	int32_t err;
 
-	no_os_gpio_set_value(desc->ldac, NO_OS_GPIO_LOW);
-	no_os_udelay(AD3552R_LDAC_PULSE_US);
-	no_os_gpio_set_value(desc->ldac, NO_OS_GPIO_HIGH);
+	if (!desc->ldac) {
+		if (is_fast)
+			return ad3552r_write_reg(desc,
+						 AD3552R_REG_ADDR_SW_LDAC_16B,
+						 mask);
+		else
+			return ad3552r_write_reg(desc,
+						 AD3552R_REG_ADDR_SW_LDAC_24B,
+						 mask);
+	} else {
+		no_os_gpio_set_value(desc->ldac, NO_OS_GPIO_LOW);
+		if (NO_OS_IS_ERR_VALUE(err))
+			return err;
 
+		no_os_udelay(AD3552R_LDAC_PULSE_US);
+		no_os_gpio_set_value(desc->ldac, NO_OS_GPIO_HIGH);
```

Comment:
ditto: error check and handling.

Reply:
Where do you assign the return value of `no_os_gpio_set_value(desc->ldac, NO_OS_GPIO_HIGH);` to `err` variable?